backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/chip/BpkChip.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.takeOrElse
+import androidx.compose.ui.text.style.TextOverflow
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.CloseCircle
+import net.skyscanner.backpack.compose.tokens.Tick
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+enum class BpkChipState {
+  Off,
+  On,
+  Disabled,
+}
+
+enum class BpkChipStyle {
+  Default,
+  OnDark,
+}
+
+enum class BpkChipType {
+  Option,
+  Select,
+  Dismiss,
+}
+
+@Composable
+fun BpkChip(
+  text: String,
+  modifier: Modifier = Modifier,
+  state: BpkChipState = BpkChipState.Off,
+  style: BpkChipStyle = BpkChipStyle.Default,
+  icon: BpkIcon? = null,
+  type: BpkChipType = BpkChipType.Option,
+  onClick: (BpkChipState) -> Unit = {},
+) {
+
+  val backgroundColor by animateColorAsState(targetValue = state.backgroundColor.takeOrElse { style.backgroundColor })
+  val contentColor by animateColorAsState(targetValue = state.contentColor.takeOrElse { style.contentColor })
+
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Md),
+    modifier = modifier
+      .height(BpkSpacing.Xl)
+      .clip(ChipShape)
+      .background(backgroundColor)
+      .selectable(
+        selected = state == BpkChipState.On,
+        enabled = state != BpkChipState.Disabled,
+        onClick = {
+          when (state) {
+            BpkChipState.Off -> onClick(BpkChipState.On)
+            BpkChipState.On -> onClick(BpkChipState.Off)
+            BpkChipState.Disabled -> Unit // do nothing
+          }
+        }
+      )
+      .padding(horizontal = BpkSpacing.Base),
+  ) {
+
+    if (icon != null) {
+      BpkIcon(
+        icon = icon,
+        size = BpkIconSize.Small,
+        contentDescription = null,
+        tint = contentColor,
+      )
+    }
+
+    BpkText(
+      text = text,
+      color = contentColor,
+      style = BpkTheme.typography.footnote,
+      maxLines = 1,
+      overflow = TextOverflow.Ellipsis,
+    )
+
+    val trailingIcon = type.icon
+    if (trailingIcon != null) {
```

Comment:
yeah we probably need an extra paramenter

Reply:
On the other side, if you add this parameter it'll just be added to chip text. Something like "Selected. 'Dismiss' 'dismiss'. Double-tap to toggle". And while the icon is different, the semantics of usage remains the same "selected/not selected, disabled/double-tap to activate", and the text provides enough context to understand what's going to be triggered. It's hard to find the other usages, rather than a description of context or semantics, and it introduces extra complexity to the API.

Reply:
It might be worth checking with adam & heather what the best approach is for this. If there's a visual distinction it feels important for accessibility as well.

Reply:
The visual distinction does not always mean accessibility importance – for instance, we don't announce the icons on the buttons. But a check won't harm!